demos/asm/measurereadlatency_x86.S

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+.intel_syntax noprefix
+
+.global MeasureReadLatency
+// uint64_t MeasureReadLatency(const char* address);
+//
+// See measurereadlatency_x86_64.S for more details on what this function does,
+// in particular for why MFENCEs and LFENCEs occur where they do. The only
+// comments here are specific to the 32-bit implementation.
+MeasureReadLatency:
+  // Prologue
+  push ebp
```

Comment:
Is this necessary? Cannot you use esp straight ahead as if you compiled with -fomit-frame-pointer or /Oy? The same for the win32 assembly.

Reply:
Admittedly I didn't put too much thought into this, but thinking about it now I feel like the balance is in favor of setting up a stack frame:
1. This code might be compiled into a binary with or without FPO. If we use a frame pointer it's at worst a little slower; if we *don't* use a frame pointer we could totally break the binary's assumptions.
2. Frame pointers make debugging easier. (Compilers need to generate more complicated stack walking metadata in the FPO case that we're not generating in this assembly.)
3. Having a base pointer is _really_ convenient when hand-writing assembly. Otherwise if we added another stack operation we'd have to remember to update the `esp` offset we use to read `address` off the stack.